packages/components/src/components/dropdown/dropdown.js

```diff
@@ -82,6 +88,9 @@ class Dropdown extends mixin(
    * Opens and closes the dropdown menu.
    * @param {Event} [event] The event triggering this method.
    */
+  // @question this function is called every time I click on the document and executes for
```

Comment:
IIRC this is for click-outside or focus-outside. Given some of the logic is not relevant to that, we may split the logic, one for global document listener, another for root element listener.

Reply:
@asudoh I feel like this is beyond the scope of this PR but do you want me to open up an issue to look in to refactoring this?

Reply:
Sounds good to me - Thanks @elizabethsjudd!